OpenGL/Core/GlMethods.cs

```diff
@@ -723,7 +723,10 @@ public static void BindBuffer<T>(VBO<T> buffer)
         public static void BindBufferToShaderAttribute<T>(VBO<T> buffer, ShaderProgram program, string attributeName)
             where T : struct
         {
-            uint location = (uint)Gl.GetAttribLocation(program.ProgramID, attributeName);
+            var cachedAttribute = program[attributeName];
+            if(cachedAttribute == null)
+                throw new ArgumentException($"{attributeName} did not exist in the shader.", nameof(attributeName));
```

Comment:
On second though, I think it is best to avoid throwing an exception here.  If the cachedAttribute doesn't exist we should just fallback to the old method of calling `Gl.GetAttribLocation(program.ProgramID, attributeName);`.  Throwing the exception introduces a possible regression for some use cases/projects.

Reply:
All right, if `cachedAttribute` is null, it will just use `Gl.GetAttribLocation`.
https://github.com/giawa/opengl4csharp/pull/39/commits/9ce0d030838443d2d604f70f69cf077f9eb27863